OpenRPA.NativeMessagingHost/myAddon2/background.js

```diff
@@ -974,3 +974,53 @@ async function downloadsOnChanged(delta) {
     if (port != null) port.postMessage(JSON.parse(JSON.stringify(message)));
 
 }
+
+async function getFocusedTab() {
+    let [tab] = await tabsquery({ active: true, lastFocusedWindow: true });
+    return tab;
+}
+
+function getCurrentBrowser() {
+    if (isChrome) return "chrome";
+    if (isFirefox) return "ff";
+    if (isChromeEdge) return "edge";
+}
+
+function pushNavigateEvent(functionName, data, event) {
+    var message = {
+        functionName: functionName,
+        browser: data.browser,
+        windowId: data.tab.windowId,
+        tabid: data.tab.id,
+        tab: data.tab,
+        result: JSON.stringify(event)
+    };
+
+    console.debug(`[send] ${message.functionName}`);
+    if(port!=null) port.postMessage(JSON.parse(JSON.stringify(message)));
+}
+
+async function handleNavigation(event) { 
+    var tab = await getFocusedTab();
+    var browser = getCurrentBrowser();
+    var data = {
+        browser: browser,
+        tab: tab
+    }
+    
+    pushNavigateEvent("navigate", data, event);
+}
+
+async function handleOnCommittedNavigation(event) { 
+    var filteredTransitions = ["typed", "auto_bookmark"];
+    if (event.frameType == 'outermost_frame' && filteredTransitions.includes(event.transitionType)) 
+        handleNavigation(event);
+}
+
+async function OnPageNavigation() {
+    const filter = { url: [{ urlMatches: 'https://*/*', }] };
```

Comment:
should it only accept `https`?

Reply:
Nope. We can accept both. Done.

Reply:
I changed the approach from `*://*/*` to `://*/*` Cause the first one was producing a bug.